### ISSUE TITLE
Could this app work on Linux?

### ISSUE THREAD

**Issue by leillo1975** [OpenSimTools/AMS2CM#96]
Automobilista 2 works on Linux without problems even since its release in Early Access thanks to Steam Play-Proton/Wine, and has awesome performance, which is why many players of our operating system enjoy it. What your software allows is amazing and we would also like to use it. Is it possible to make a version of CM for GNU/Linux?

**Comment by paoloambrosio** [comment]
AFAIK AMS2 on Linux runs the Windows binaries via the [Proton](https://github.com/ValveSoftware/Proton) compatibility layer.

Someone should try to run AMS2CM on [Wine](https://www.winehq.org/) (or Proton).

**Comment by paoloambrosio** [comment]
## Running on Wine/Proton

Currently this would be my preferred option, due to complications migrating to a Linux-native solution. The following is a brain dump of the issues that might arise when running on Wine...

### Core/CLI
- I expect it not to be able to find the game directory. CM runs on Wine, but Steam is built natively for Linux so it won't be found by CM. Workaround is to specify the game directory in config.
- CM uses two external libraries that are a black box: 7z and oodle. The second one is only called for bootfiles, so it will bypassed for skins.

### GUI
- WinUI 3 probably requires some Windows features that are not implemented in Wine.

## Linux Native Build

### Core/CLI
- CM currently looks in the registry to find where Steam is installed. We would have to find a way to find the same information on Linux.
- [7zip](https://www.7-zip.org/) is distributed as LGPL for Windows and Linux. Currently we rely on a wrapper library. I'm not sure how calling a native library from .NET works on Linux.
- [Oodle](http://www.radgametools.com/oodle.htm) is a proprietary library used by the Madness Engine and licensed for inclusion in AMS2. CM uses the one present in the game's directory for bootfile extraction. Bootfiles are still needed for tracks (thanks to Reiza). Since AMS2 runs on Proton, no Linux native version of this library is available to us. There is a crazy [open-source project](https://github.com/McSimp/linoodle) to provide a wrapper around it, that we could consider, though not a clean solution.

### GUI

- [WinUI 3](https://learn.microsoft.com/en-us/windows/apps/winui/winui3/) targets Windows 10 and later. Microsoft has tried several times to create a new UI framework after WinForms, and failed miserably every time, so I wouldn't be too sad to put WinUI in the bin. [.NET MAUI](https://dotnet.microsoft.com/en-us/apps/maui) would sound promising, but it doesn't come with Linux Desktop support: so much for multi-platform 🤯 Open-source alternatives to the Microsoft mess (e.g. [Avalonia UI](https://avaloniaui.net/)) seem to be a better option.

**Comment by leillo1975** [comment]
The first problem I have is that I can't install .NET6 either with protontricks or with the Windows installer....

**Comment by paoloambrosio** [comment]
I was able to run AMS2CM's CLI without any problem on Debian 12 with Wine 8.
- Installed Wine 8 (`sudo apt install wine64 wine32:i386`)
- Executed `wincfg` and selected "Windows 10" (not sure if it's relevant)
- Installed .NET 6 Runtime x64 (`wine windowsdesktop-runtime-6.0.22-win-x64.exe`)
- Installed AMS2CM 0.2.1 (`wine AMS2CM.msi`) including the CLI
- Created a fake installation of AMS2 under `.local/share/Steam/steamapps/common/Automobilista 2` copying the game files that AMS2CM requires (don't have 100GB space to install it!)
- Changed `Config.yaml` adding the full game path (`Z:\home\vboxuser\.local\share\Steam\steamapps\common\Automobilista 2`)
- Added a mod inside `.local/share/Steam/steamapps/common/Automobilista 2/Mods/Enabled`
- Executed CM (`wine ~/.wine/drive_c/Program Files/AMS2CM/AMS2CM.exe`)

**Comment by leillo1975** [comment]
The third step was where I failed. I'll try again. Thank you very much for taking the time to test this.

**Comment by claiton-rocha** [comment]
Eu consegui instalar através do "Bottles". 
I can install with "Bottles".
Sorry my bad english.


Create a new bottle

![Captura de tela de 2024-07-21 10-55-22](https://github.com/user-attachments/assets/9660ac00-46ff-49aa-be6c-807e7e4d37cc)


Install using "Start executable"

![Captura de tela de 2024-07-21 10-57-39](https://github.com/user-attachments/assets/be6b4a3d-508f-4ba2-915a-d9e8266f140b)


Install dependencies

![Captura de tela de 2024-07-21 10-57-392](https://github.com/user-attachments/assets/5a349fff-125b-46b6-a012-38ae090f08bc)

![Captura de tela de 2024-07-21 11-05-13](https://github.com/user-attachments/assets/cb0ce929-7681-496b-833e-757d194703ab)

How I use another disc for games, create a new unity for AMS 2 with "bottles"
Can use the same thing for same drive, your choice

![Captura de tela de 2024-07-21 10-57-393](https://github.com/user-attachments/assets/57d48d0d-ad83-47d4-9885-49537850202c)

![Captura de tela de 2024-07-21 11-12-56](https://github.com/user-attachments/assets/f72417e9-99af-4154-95b9-ae3383037d71)

Setting the path in config.yaml

![Captura de tela de 2024-07-21 11-16-22](https://github.com/user-attachments/assets/795c3515-9b48-411c-ac04-3cfdb9f64cf9)

Finally execute ASM2CM in terminal, note: you have access ASM2CM.exe using MS commands (dir, cd, etc...)

![Captura de tela de 2024-07-21 11-18-14](https://github.com/user-attachments/assets/8993aad2-cd1a-46ab-aa1e-7822aa783701)

I hope I was able to help. Sorry for my interface being in Portuguese but nothing that Google Translate can't help with.

**Comment by leillo1975** [comment]
Thanks for the info. Let's try it tonight!

**Comment by paoloambrosio** [comment]
> what are the key files of AMS2 that the CM needs?

I think only these are required:
```
Pakfiles\BOOTFLOW.bff
Pakfiles\BOOTSPLASH.bff
Pakfiles\PHYSICSPERSISTENT.bff
```

I have also added these, but I don't think that they are necessary:
```
AMS2AVX.dll
AMS2AVX.exe
AMS2AVX.runtimeconfig.json
```

> Config.yaml--is this the file in my Steam install

This is in the AMS2CM install directory.

> I've tried to start AMS2CM with Wine 10 many times--I always get the message that .Net 6.0.27 is needed, and I am asked if I would like to download it, but it never does. I tried installing through Winetricks, too.

.NET is indeed required. Download ".NET Desktop Runtime" x64 [from Microsoft](https://dotnet.microsoft.com/en-us/download/dotnet/6.0) and install it with Wine (as detailed in a [previous message](https://github.com/OpenSimTools/AMS2CM/issues/96#issuecomment-1727748305)).

**Comment by a1872** [comment]
Hi.  I was wondering what was meant by "full game path"

"Changed Config.yaml adding the full game path (Z:\home\vboxuser\.local\share\Steam\steamapps\common\Automobilista 2)"

What about the drive letter--I  think I used drive c in the wine install of the CM, as the next sentence shows.

"/home/anthony/Games/ams2cm/drive_c/Program Files/AMS2CM"--is this what is meant?  It isn't the directory created by myself in the previous step:

"Created a fake installation of AMS2 under .local/share/Steam/steamapps/common/Automobilista 2 copying the game files that AMS2CM requires (don't have 100GB space to install it!)" 

I started creating the path at /Steam/steamapps/etc.

The real Steam installation is on an external Seagate drive--by saying "full game", I was just wondering if you meant this by chance?  Thanks.

**Comment by paoloambrosio** [comment]
Yes, the full game path includes the drive letter, from the Wine/Windows point of view (e.g. `Z:\home\vboxuser.local\share\Steam\steamapps\common\Automobilista 2`). Of course, if you have a real AMS2 installation, you don't need to create a fake one; I only did it to prove that it was working.

If you have more questions, it's better if we continue this conversation on [Discord](https://discord.gg/F7kMHSf3XR).
